Core/src/main/java/com/plotsquared/core/location/UncheckedWorldLocation.java

```diff
@@ -0,0 +1,75 @@
+/*
+ *       _____  _       _    _____                                _
+ *      |  __ \| |     | |  / ____|                              | |
+ *      | |__) | | ___ | |_| (___   __ _ _   _  __ _ _ __ ___  __| |
+ *      |  ___/| |/ _ \| __|\___ \ / _` | | | |/ _` | '__/ _ \/ _` |
+ *      | |    | | (_) | |_ ____) | (_| | |_| | (_| | | |  __/ (_| |
+ *      |_|    |_|\___/ \__|_____/ \__, |\__,_|\__,_|_|  \___|\__,_|
+ *                                    | |
+ *                                    |_|
+ *            PlotSquared plot management system for Minecraft
+ *               Copyright (C) 2014 - 2022 IntellectualSites
+ *
+ *     This program is free software: you can redistribute it and/or modify
+ *     it under the terms of the GNU General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     (at your option) any later version.
+ *
+ *     This program is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ *
+ *     You should have received a copy of the GNU General Public License
+ *     along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+package com.plotsquared.core.location;
+
+import com.plotsquared.core.util.AnnotationHelper;
+import com.sk89q.worldedit.math.BlockVector3;
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+/**
+ * Used internally for generation to reference locations in worlds that "don't exist yet". There is no guarantee that the world
+ * name provided by {@link UncheckedWorldLocation#getWorldName()} exists on the server.
+ *
+ * @since TODO
+ */
+@AnnotationHelper.ApiDescription(info = "Internal use only. Subject to changes at any time.")
```

Comment:
Probably makes sense to apply that on the public methods too, IJ isn't very generous with propagating class annotations on methods when browsing source jars.